client/my-sites/term-tree-selector/terms.jsx

```diff
@@ -0,0 +1,422 @@
+			count += this.props.termsHierarchy.length;
+		}
+
+		if ( ! this.props.lastPage || this.hasNoSearchResults() || this.props.loading ) {
+			count += 1;
+		}
+
+		return count;
+	},
+
+	onSearch( event ) {
+		const searchTerm = event.target.value;
+		if ( this.state.searchTerm && ! searchTerm ) {
+			this.props.onSearch( '' );
+		}
+
+		if ( searchTerm === this.state.searchTerm ) {
+			return;
+		}
+
+		if ( ! this.hasPerformedSearch ) {
+			this.hasPerformedSearch = true;
+			analytics.ga.recordEvent( this.props.analyticsPrefix, 'Performed Term Search' );
+		}
+
+		this.setState( { searchTerm } );
+		this.debouncedSearch();
+	},
+
+	getSelectedIds( selected ) {
+		const selectedObjects = selected || this.props.selected;
+		return selectedObjects.map( function( item ) {
+			if ( ! item.ID ) {
+				return item;
+			}
+
+			return item.ID;
+		} );
+	},
+
+	renderItem( item ) {
+		const onChange = ( ...args ) => this.props.onChange( item, ...args );
+		const setItemRef = ( ...args ) => this.setItemRef( item, ...args );
+
+		const { multiple, defaultTermId, translate } = this.props;
+		const selectedIds = this.getSelectedIds();
+		const itemId = item.ID;
+		const name = unescapeString( item.name ) || translate( 'Untitled' );
+		const checked = includes( selectedIds, itemId );
+		const inputType = multiple ? 'checkbox' : 'radio';
+		let disabled;
+
+		if ( multiple && checked && defaultTermId &&
+				( 1 === selectedIds.length ) &&
+				( defaultTermId === itemId ) ) {
+			disabled = true;
+		}
+
+		const input = (
+			<input
+				type={ inputType }
+				value={ itemId }
+				onChange={ onChange }
+				disabled={ disabled }
+				checked={ checked }
+			/>
+		);
+
+		return (
+			<div
+				key={ itemId }
+				ref={ setItemRef }
+				className="term-tree-selector__list-item">
+				<label>
+					{ input }
+					<span className="term-tree-selector__label">{ name }</span>
+				</label>
+				{ item.items && (
+					<div className="term-tree-selector__nested-list">
+						{ item.items.map( this.renderItem ) }
+					</div>
+				) }
+			</div>
+		);
+	},
+
+	renderRow( { index } ) {
+		if ( this.hasNoSearchResults() || this.hasNoTerms() ) {
+			return (
+				<div key="no-results" className="term-tree-selector__list-item is-empty">
+					{ ( this.hasNoSearchResults() || ! this.props.emptyMessage ) && (
+						<NoResults
+							createLink={ this.props.createLink } />
+					) }
+					{ this.hasNoTerms() && this.props.emptyMessage }
+				</div>
+			);
+		}
+
+		const item = this.getItem( index );
+		if ( item ) {
+			return this.renderItem( item );
+		}
+
+		if ( ! this.props.loading && this.props.lastPage ) {
+			return;
+		}
+
+		return (
+			<div key="placeholder" className="term-tree-selector__list-item is-placeholder">
+				<label>
+					<input
+						type={ this.props.multiple ? 'checkbox' : 'radio' }
+						disabled
+						className="term-tree-selector__input" />
+					<span className="term-tree-selector__label">
+						{ this.translate( 'Loading…' ) }
```

Comment:
We should be consistent with our usage of `localize` vs. the `this.translate` mixin, probably changing this to `this.props.translate`